### PR TITLE
Add option for invalid extra files targets to not fail build

### PR DESCRIPTION
**Pull request by acbdev**
**Why**
Previously, encountering invalid extra file target labels resulted in project generation failure and an error message. The new option allows for increased flexibility by providing the choice to either fail the project generation or simply emit a warning and ignore the invalid label.

This is especially helpful when a target is excluded from the dependency graph but still passed in associated_extra_files.

**What**
- Created option called `fail_for_invalid_extra_files_targets`. This determines wether, when processing targets, invalid extra files without labels will fail or just emit a warning

### CODE REVIEW

**Comment by brentleyjones** [comment]
You'll need to sign-off on your commit(s) (squash then it's only one commit) for the DCO check.

**Comment by brentleyjones** [comment]
You'll also need to update the docs (`./docs/update_docs.sh`).

Can you add an example of this to `examples/integrations` as well? Will need to adjust a constant for the bool in `examples/integrations/xcodeproj_targets.bzl` and use it in `examples/integration/BUILD` and `examples/integration/test/fixtures/BUILD`. Then you'll need to update the fixtures (`./test/update_all_fixtures.sh`)

File: examples/integration/test/fixtures/bazel-output-base/DO_NOT_BUILD_HERE
Comment (brentleyjones) [inline examples/integration/test/fixtures/bazel-output-base/DO_NOT_BUILD_HERE:1]: `examples/integration/test/fixtures/bazel-output-base` and all children shouldn't be committed.

**Comment by erikkerber** [comment]
@acbdev Mind adding the tiniest of tiny motivations in the PR description?

Useful especially when new attributes are made to make it easy to backtrack how and why they exist and any discussion around them 🙏🏼

File: examples/integration/test/fixtures/bazel-output-base/DO_NOT_BUILD_HERE
Comment (acbdev) [reply examples/integration/test/fixtures/bazel-output-base/DO_NOT_BUILD_HERE:1]: Oh ok sorry I did not realise!! I will remove them

**Review by brentleyjones — APPROVED** [review]
(Approving to run CI.)

File: xcodeproj/internal/xcodeproj_rule.bzl [hunk 1]
```diff
@@ -1801,6 +1819,9 @@ def make_xcodeproj_rule(
             mandatory = True,
         ),
         "default_xcode_configuration": attr.string(),
+        "fail_for_invalid_extra_files_targets": attr.bool(
+            default = True,
+        ),
```
[the remark this answers is not in the document]
Comment (brentleyjones) [reply xcodeproj/internal/xcodeproj_rule.bzl]: It would, but the effort to create the linter probably outweighs the investment.

**Review by brentleyjones — APPROVED** [review]

**Review by brentleyjones — APPROVED** [review]

**Review by brentleyjones — APPROVED** [review]

**Comment by brentleyjones** [comment]
Thanks @acbdev!

**Pull request merged**